src/scheduler/fifo.cpp

```diff
@@ -1646,7 +1653,10 @@ run_update_resresv(status *policy, int pbs_sd, server_info *sinfo,
 					fflush(stdout);
 #endif /* localmod 031 */
 
-					pbsrc = run_job(get_svr_inst_fd(pbs_sd, rr->job->svr_inst_id), rr, execvnode, sinfo->has_runjob_hook, err);
+					if (ns && ns[0])
+						node_owner = ns[0]->ninfo->svr_inst_id;
```

Comment:
I think the cost should be negligible. Currently, the scheduler will check whether it has to send the node owner. The server will do the check at its end as other clients could send in either case.
In case if we are sending the node_owner we also need to frame the extend with run_on:server format. If that cost is equivalent to the strcmp then sending over the wire will be the deciding factor.